src/scrapers/beinleumi.js

```diff
@@ -0,0 +1,265 @@
+const DATE_FORMAT = 'DD/MM/YYYY';
+const NO_TRANSACTION_IN_DATE_RANGE_TEXT = 'לא נמצאו נתונים בנושא המבוקש';
+const DATE_COLUMN_CLASS = 'date first';
+const DESCRIPTION_COLUMN_CLASS = 'reference wrap_normal';
+const REFERENCE_COLUMN_CLASS = 'details';
+const DEBIT_COLUMN_CLASS = 'debit';
+const CREDIT_COLUMN_CLASS = 'credit';
+const ERROR_MESSAGE_CLASS = 'NO_DATA';
+const ACCOUNTS_NUMBER = 'div.fibi_account span.acc_num';
+const CLOSE_SEARCH_BY_DATES_BUTTON_CLASS = 'ui-datepicker-close';
+const SHOW_SEARCH_BY_DATES_BUTTON_VALUE = 'הצג';
+const TRANSACTIONS_TABLE = 'table[id*=\'dataTable\']';
+const TRANSACTIONS_PAGINATION_LIMIT = 20;
+const NEXT_PAGE_LINK = 'a#Npage.paging';
+const CLICK_ELEMENT_NOT_EXISTS_ERROR_MSG = 'Cannot read property';
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = [/FibiMenu\/Online/];
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = [/FibiMenu\/Marketing\/Private\/Home/];
+  return urls;
+}
+
+function createLoginFields(credentials) {
+  return [
+    { selector: '#username', value: credentials.username },
+    { selector: '#password', value: credentials.password },
+  ];
+}
+
+function getAmountData(amountStr) {
+  const amountStrCopy = amountStr.replace(',', '');
+  return parseFloat(amountStrCopy);
+}
+
+function getTxnAmount(txn) {
+  const credit = getAmountData(txn.credit);
+  const debit = getAmountData(txn.debit);
+  return (Number.isNaN(credit) ? 0 : credit) - (Number.isNaN(debit) ? 0 : debit);
+}
+
+function convertTransactions(txns) {
+  return txns.map((txn) => {
+    const convertedDate = moment(txn.date, DATE_FORMAT).toISOString();
+    const convertedAmount = getTxnAmount(txn);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: txn.reference ? parseInt(txn.reference, 10) : null,
+      date: convertedDate,
+      processedDate: convertedDate,
+      originalAmount: convertedAmount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: convertedAmount,
+      status: txn.status,
+      description: txn.description,
+      memo: txn.memo,
+    };
+  });
+}
+
+function getTransactionDate(tds, transactionsColsTypes) {
+  return (tds[transactionsColsTypes[DATE_COLUMN_CLASS]] || '').trim();
+}
+
+function getTransactionDescription(tds, transactionsColsTypes) {
+  return (tds[transactionsColsTypes[DESCRIPTION_COLUMN_CLASS]] || '').trim();
+}
+
+function getTransactionReference(tds, transactionsColsTypes) {
+  return (tds[transactionsColsTypes[REFERENCE_COLUMN_CLASS]] || '').trim();
+}
+
+function getTransactionDebit(tds, transactionsColsTypes) {
+  return (tds[transactionsColsTypes[DEBIT_COLUMN_CLASS]] || '').trim();
+}
+
+function getTransactionCredit(tds, transactionsColsTypes) {
+  return (tds[transactionsColsTypes[CREDIT_COLUMN_CLASS]] || '').trim();
+}
+
+function extractTransactionDetails(txnRow, transactionsColsTypes) {
+  const tds = txnRow.innerTds;
+  return {
+    status: TRANSACTION_STATUS.COMPLETED,
```

Comment:
it seems that you are always returning status `COMPLETED`. are you familiar with `PENDING` status? it usually happens on holidays and weekends. Do you know if BeinLeumi has pending transactions?

Reply:
I'm not familiar with PENDING transactions on the Beinleumi and I think they show completed transactions only.

Reply:
I think all banks show PENDING transactions, try to check your account on Tuesday / Wednesday. maybe will will get lucky and you will have pending transactions then :)

Reply:
@eshaham  and @esakal , 
Please **do not** approve this PR yet.

I've decided to investigate Eran's assumption on the bank website, so I've made a money movement between my accounts (from different banks).
I've fountd out that the transaction of the money movement I've just made appears on the bank website in another table under the title ״תנועות מהיום״.

The problem in the source code is that both of the transactions tables have non meaningful ids (dataTable077 and dataTable023) and it just means that **I have a bug** in my code which now extract transactions from both tables and set them as COMPLETED transactions.

Please allow me to fix that bug first, and to support PENDING transactions as well.

Reply:
Hi @eshaham and @esakal,
I've added a support of pending transactions on the last commit and I think the script is ready now.
Will be happy to hear your opinion.